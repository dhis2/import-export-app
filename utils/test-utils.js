import * as testingLibrary from '@testing-library/react'
import * as customQueries from './custom-queries.js'

const { queries } = testingLibrary

console.log('testingLibrary', testingLibrary)

const customRender = (ui, options) =>
    testingLibrary.render(ui, {
        queries: { ...queries, ...customQueries },
        ...options,
    })

/* eslint-disable */
// re-export everything
export * from '@testing-library/react'

// override render method
export { customRender as render }
/* eslint-enable */
