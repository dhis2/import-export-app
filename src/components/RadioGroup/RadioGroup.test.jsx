import React from 'react'
import { render, fireEvent, cleanup } from 'test-utils'
import '@testing-library/jest-dom/extend-expect.js'

import { RadioGroup } from '../index.js'

const options = [
    { value: 'a', label: 'Abc' },
    { value: 'b', label: 'Def' },
    { value: 'c', label: 'Ghi' },
]

const setValue = jest.fn().mockName('setValue')

const props = {
    name: 'format',
    label: 'Format',
    options: options,
    checked: options[0].value,
    setValue: setValue,
    dataTest: 'radio-group-format',
}

afterEach(() => {
    cleanup()
    jest.clearAllMocks()
})

test('clicking on an option triggers setValue', () => {
    const { getByDataTest } = render(<RadioGroup {...props} />)
    const def = getByDataTest('radio-group-format-b')
    expect(def).toHaveTextContent('Def')
    fireEvent.click(def)
    expect(setValue).toHaveBeenCalledTimes(1)
    expect(setValue).toHaveBeenCalledWith('b')
})

it('matches snapshot', () => {
    const { asFragment } = render(<RadioGroup {...props} />)
    expect(asFragment()).toMatchSnapshot()
})
