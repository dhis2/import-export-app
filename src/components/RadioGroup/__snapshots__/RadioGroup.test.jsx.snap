// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
    data-test="radio-group-format"
  >
    <span
      class="label"
    >
      Format
    </span>
    <div
      class="horizontal"
    >
      <div>
        <label
          class="jsx-2199927085 "
          data-test="radio-group-format-a"
        >
          <input
            checked=""
            class="jsx-2199927085"
            name="format"
            type="radio"
            value="a"
          />
          <div
            class="jsx-2199927085 icon"
          >
            <svg
              class="jsx-1260643562 jsx-1120332023 checked"
              viewBox="0 0 18 18"
              xmlns="http://www.w3.org/2000/svg"
            >
              <circle
                class="jsx-1260643562 jsx-1120332023 background"
                cx="9"
                cy="9"
                r="9"
              />
              <path
                class="jsx-1260643562 jsx-1120332023 outer unchecked"
                d="M9,0 C13.9705627,0 18,4.02943725 18,9 C18,13.9705627 13.9705627,18 9,18 C4.02943725,18 0,13.9705627 0,9 C0,4.02943725 4.02943725,0 9,0 Z M9,1 C4.581722,1 1,4.581722 1,9 C1,13.418278 4.581722,17 9,17 C13.418278,17 17,13.418278 17,9 C17,4.581722 13.418278,1 9,1 Z"
              />
              <path
                class="jsx-1260643562 jsx-1120332023 outer checked"
                d="M9,18 C13.9705627,18 18,13.9705627 18,9 C18,4.02943725 13.9705627,0 9,0 C4.02943725,0 0,4.02943725 0,9 C0,13.9705627 4.02943725,18 9,18 Z M9,16 C5.13400675,16 2,12.8659932 2,9 C2,5.13400675 5.13400675,2 9,2 C12.8659932,2 16,5.13400675 16,9 C16,12.8659932 12.8659932,16 9,16 Z"
              />
              <circle
                class="jsx-1260643562 jsx-1120332023 inner"
                cx="9"
                cy="9"
                r="5"
              />
            </svg>
          </div>
          Abc
        </label>
      </div>
      <div>
        <label
          class="jsx-2199927085 "
          data-test="radio-group-format-b"
        >
          <input
            class="jsx-2199927085"
            name="format"
            type="radio"
            value="b"
          />
          <div
            class="jsx-2199927085 icon"
          >
            <svg
              class="jsx-1260643562 jsx-1120332023 "
              viewBox="0 0 18 18"
              xmlns="http://www.w3.org/2000/svg"
            >
              <circle
                class="jsx-1260643562 jsx-1120332023 background"
                cx="9"
                cy="9"
                r="9"
              />
              <path
                class="jsx-1260643562 jsx-1120332023 outer unchecked"
                d="M9,0 C13.9705627,0 18,4.02943725 18,9 C18,13.9705627 13.9705627,18 9,18 C4.02943725,18 0,13.9705627 0,9 C0,4.02943725 4.02943725,0 9,0 Z M9,1 C4.581722,1 1,4.581722 1,9 C1,13.418278 4.581722,17 9,17 C13.418278,17 17,13.418278 17,9 C17,4.581722 13.418278,1 9,1 Z"
              />
              <path
                class="jsx-1260643562 jsx-1120332023 outer checked"
                d="M9,18 C13.9705627,18 18,13.9705627 18,9 C18,4.02943725 13.9705627,0 9,0 C4.02943725,0 0,4.02943725 0,9 C0,13.9705627 4.02943725,18 9,18 Z M9,16 C5.13400675,16 2,12.8659932 2,9 C2,5.13400675 5.13400675,2 9,2 C12.8659932,2 16,5.13400675 16,9 C16,12.8659932 12.8659932,16 9,16 Z"
              />
              <circle
                class="jsx-1260643562 jsx-1120332023 inner"
                cx="9"
                cy="9"
                r="5"
              />
            </svg>
          </div>
          Def
        </label>
      </div>
      <div>
        <label
          class="jsx-2199927085 "
          data-test="radio-group-format-c"
        >
          <input
            class="jsx-2199927085"
            name="format"
            type="radio"
            value="c"
          />
          <div
            class="jsx-2199927085 icon"
          >
            <svg
              class="jsx-1260643562 jsx-1120332023 "
              viewBox="0 0 18 18"
              xmlns="http://www.w3.org/2000/svg"
            >
              <circle
                class="jsx-1260643562 jsx-1120332023 background"
                cx="9"
                cy="9"
                r="9"
              />
              <path
                class="jsx-1260643562 jsx-1120332023 outer unchecked"
                d="M9,0 C13.9705627,0 18,4.02943725 18,9 C18,13.9705627 13.9705627,18 9,18 C4.02943725,18 0,13.9705627 0,9 C0,4.02943725 4.02943725,0 9,0 Z M9,1 C4.581722,1 1,4.581722 1,9 C1,13.418278 4.581722,17 9,17 C13.418278,17 17,13.418278 17,9 C17,4.581722 13.418278,1 9,1 Z"
              />
              <path
                class="jsx-1260643562 jsx-1120332023 outer checked"
                d="M9,18 C13.9705627,18 18,13.9705627 18,9 C18,4.02943725 13.9705627,0 9,0 C4.02943725,0 0,4.02943725 0,9 C0,13.9705627 4.02943725,18 9,18 Z M9,16 C5.13400675,16 2,12.8659932 2,9 C2,5.13400675 5.13400675,2 9,2 C12.8659932,2 16,5.13400675 16,9 C16,12.8659932 12.8659932,16 9,16 Z"
              />
              <circle
                class="jsx-1260643562 jsx-1120332023 inner"
                cx="9"
                cy="9"
                r="5"
              />
            </svg>
          </div>
          Ghi
        </label>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
